pkg/apis/monitoring/v1/types.go

```diff
@@ -1373,6 +1373,9 @@ type PodMetricsEndpoint struct {
 	FollowRedirects *bool `json:"followRedirects,omitempty"`
 	// Whether to enable HTTP2.
 	EnableHttp2 *bool `json:"enableHttp2,omitempty"`
+	// Drop pods that are not running. (Failed, Succeeded, Unknown).
+	// More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+	FilterRunning *bool `json:"filterRunning,omitempty"`
```

Comment:
@SuperQ do we want to comment that this is enabled by default? 

Also wondering that since it is enabled by default does it make sense to be a pointer? Perhaps alternatively we can use the `+kubebuilder:default` annotation?

Reply:
Sure, I can do that.

As for the pointer, I wasn't quite sure how to make a bool true by default here.

Reply:
I'm not sure that `+kubebuilder:default` will play nice with a boolean field (`omitempty` might also come in the way) but it's worth checking.

Reply:
Yea, I was tinkering around with this a bit and couldn't get it to work. The code generator always wants to make bools into pointers.

Reply:
From the [docs](https://book.kubebuilder.io/reference/markers/crd-validation.html)

```
// +kubebuilder:default
any
sets the default value for this field.
A default value will be accepted as any value valid for the field. Formatting for common types include: boolean: true, string: Cluster, numerical: 1.24, array: {1,2}, object: {policy: &#34;delete&#34;}). Defaults should be defined in pruned form, and only best-effort validation will be performed. Full validation of a default requires submission of the containing CRD to an apiserver.
```

but yeah, lets not get hung up on this, once we document the default value I think we are good.